### PR TITLE
Flush changes from 1.0-FTY-master into 1.0-FTY "stable"

### PR DESCRIPTION
**Pull request by jimklimov**

### CODE REVIEW

**Review by jana-rapava — DISMISSED** [review]

File: src/mlm_client_engine.inc [hunk 1]
```diff
@@ -1723,7 +1723,11 @@ int
 mlm_client_set_consumer (mlm_client_t *self, const char *stream, const char *pattern)
 {
     assert (self);
-
+    // https://github.com/zeromq/malamute/issues/217
+    // problem  : if pattern "*" is used, the related stream will be broken and
+    //            mlm broker will start overcosumming 100% CPU forever
```
Comment (aquette) [inline src/mlm_client_engine.inc]: @jimklimov double typo for upstream: overco*n*sum**ing (2nd 'm' suppressed)

**Comment by aquette** [comment]
@jimklimov could you plz confirm that this was discussed and tested (originating from) with Dgé? and thus is mergeable with no regression...

**Comment by jimklimov** [comment]
The 1fbb2b5b4f5c commit (current state of `upstream/1.0-FTY-master`) is used since end of April, that is about `snprintf()`. The earlier fix for `*` stream pattern is merged and in place for master-branch builds since early March. The `1.0-FTY` branch saves the known-stable state but is not directly used in builds, except (for weird-legacy reason) in the FTY dispatcher repo.

File: src/mlm_client_engine.inc (hunk 1 above)
Comment (jimklimov) [reply src/mlm_client_engine.inc]: thanks

**Comment by jimklimov** [comment]
As for breaking or not, per https://github.com/zeromq/malamute/issues/217 discussion the issue was to be addressed in czmq (or further in regex engine) and so not pursued in upstream malamute.

I commented my thoughts on that issue now, though.

**Comment by perrettecl** [comment]
@jimklimov and @aquette  What should we do with this PR ?

**Comment by jimklimov** [comment]
I think, if we have not yet seen definitive malamute-related regressions (with this PR state in master images for 2 months now) then it is good to merge.

**Comment by perrettecl** [comment]
Ok Thank I will validate and merger prior branching

**Review by perrettecl — APPROVED** [review]

**Pull request merged**